### PR TITLE
Add Linear+BN fusion to conversion pipeline of unit tests

### PR DESCRIPTION
**Pull request by StrycekSimon**
### Summary
Enables proper quantization and conversion of Linear+BN based models in our integration tests by adding Linear+BN fusion related passes. It also introduces Accuracy testing support for non-softmax-based models.

### Test plan
Covered by our NXP internal integration tests.

cc @robert-kalmar @JakeStevens @digantdesai

### CODE REVIEW

File: backends/nxp/tests_models/model_output_comparator.py [hunk 1]
```diff
@@ -95,17 +96,30 @@ def compare_sample(self, sample_dir, cpu_output_tensors, npu_output_tensors):
...
+        postprocess_fn: Callable[
+            [np.ndarray, str], np.ndarray
+        ] = _default_postprocess_fn,
+        tolerance=0.0,
+    ):
         """
         Comparator for comparing model prediction accuracies based on a ground-truth annotations.
         The comparator passes if finetuned model results have higher accuracy than baseline (accounting for a tolerance).
 
         :param class_dict: Dictionary mapping class names to class indices.
```
[the remark this answers is not in the document]
Comment (roman-janik-nxp) [reply backends/nxp/tests_models/model_output_comparator.py:117]: True dat.

**Review by roman-janik-nxp — COMMENTED** [review]
Please address the Copilot reviews.

File: backends/nxp/tests_models/model_output_comparator.py [hunk 2]
```diff
@@ -158,17 +175,23 @@ def compare_results(
                     dtype=torch_type_to_numpy_type(tensor_spec.dtype),
                 )
                 np.reshape(baseline_tensor, tensor_spec.shape)
+                baseline_sample_paths.append(baseline_tensor_path)
```
[the remark this answers is not in the document]
Comment (StrycekSimon) [reply backends/nxp/tests_models/model_output_comparator.py:176]: All other comparators have the same bug. Thanks for pointing out.

File: backends/nxp/tests_models/model_output_comparator.py [hunk 3]
```diff
@@ -95,17 +96,30 @@ def compare_sample(self, sample_dir, cpu_output_tensors, npu_output_tensors):
             assert np.allclose(cpu_tensor, npu_tensor, atol=self.atol)
 
 
+def _default_postprocess_fn(outputs: np.ndarray, _: str):
+    return np.argmax(outputs, axis=-1)
+
```
[the remark this answers is not in the document]
Comment (StrycekSimon) [reply backends/nxp/tests_models/model_output_comparator.py:99]: This is now resolved by summing over the batch [here](https://github.com/pytorch/executorch/pull/18527/changes#diff-ceadc24747816a2d03ca5a85028c55d947fb02ec7b012c52385cd93aff5c2159R255).

File: backends/nxp/tests_models/model_output_comparator.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (StrycekSimon) [reply backends/nxp/tests_models/model_output_comparator.py:176]: I fixed this here and in other Comparators.

File: backends/nxp/tests_models/utils.py [hunk 4]
```diff
@@ -92,12 +99,15 @@ def to_quantized_edge_program(
...
         m = prepare_qat_pt2e(module, quantizer)
+        m = AddSimulatedLinearBatchNormFusionQATPass()(m).graph_module
 
         if train_fn:
             m = move_exported_model_to_train(m)
             train_fn(m)
 
         m = move_exported_model_to_eval(m)
+        m = RemoveSimulatedLinearBatchNormFusionQATPass()(m).graph_module
+        m = FuseBatchNormWithLinearPass()(m).graph_module
```
[the remark this answers is not in the document]
Comment (StrycekSimon) [reply backends/nxp/tests_models/utils.py:110]: I resolved this by running both in the `FuseBatchNormWithLinearPass` directly.

File: backends/nxp/tests_models/model_output_comparator.py [hunk 5]
```diff
@@ -209,11 +247,15 @@ def compare_sample(
...
+            baseline_class = self.postprocess_fn(
+                baseline_tensor, baseline_filepaths[idx]
+            )
 
-            baseline_correct += baseline_class == class_id
-            finetuned_correct += finetuned_class == class_id
+            baseline_correct += sum(baseline_class == class_id)
+            finetuned_correct += sum(finetuned_class == class_id)
 
         return finetuned_correct, baseline_correct, len(baseline_output_tensors)
```
[the remark this answers is not in the document]
Comment (StrycekSimon) [reply backends/nxp/tests_models/model_output_comparator.py]: I am returning the correct total number of samples now.

File: backends/nxp/tests_models/model_output_comparator.py [hunk 6]
```diff
@@ -209,11 +247,15 @@ def compare_sample(
...
+                finetuned_tensor, finetuned_filepaths[idx]
+            )
+            baseline_class = self.postprocess_fn(
+                baseline_tensor, baseline_filepaths[idx]
+            )
 
-            baseline_correct += baseline_class == class_id
-            finetuned_correct += finetuned_class == class_id
+            baseline_correct += sum(baseline_class == class_id)
+            finetuned_correct += sum(finetuned_class == class_id)
```
[the remark this answers is not in the document]
Comment (StrycekSimon) [reply backends/nxp/tests_models/model_output_comparator.py:257]: The result should not be an integer as the input to the preprocess function should always be batched.

File: backends/nxp/tests_models/model_output_comparator.py [hunk 7]
```diff
@@ -197,7 +225,7 @@ def compare_sample(
                 f"Sample dir format invalid. Expected format: 'example_classname_0', got {sample_dir}"
             )
 
-        class_name = sample_dir.split("_")[1]
+        class_name = "_".join(sample_dir.split("_")[1:-3])
         class_id = self.inv_class_dict[class_name]
```
[the remark this answers is not in the document]
Comment (StrycekSimon) [reply backends/nxp/tests_models/model_output_comparator.py]: Resolved

File: backends/nxp/tests_models/model_output_comparator.py [hunk 8]
```diff
@@ -187,35 +208,70 @@ def compare_results(
...
+                else sum(baseline_correct)
+            )
+            finetuned_correct_total += (
+                finetuned_correct
+                if np.isscalar(finetuned_correct)
+                else sum(finetuned_correct)
+            )
+            total_samples += (
+                1 if np.isscalar(finetuned_correct) else len(baseline_correct)
+            )
```
[the remark this answers is not in the document]
Comment (StrycekSimon) [reply backends/nxp/tests_models/model_output_comparator.py:273]: We are talking about classification accuracy here, hence the name of the comparator - `ClassificationAccuracyOutputComparator`. It is a correct assumption that the predictions are 1D.

File: backends/nxp/tests_models/model_output_comparator.py [hunk 9]
```diff
@@ -157,18 +172,24 @@ def compare_results(
                     baseline_tensor_path,
                     dtype=torch_type_to_numpy_type(tensor_spec.dtype),
                 )
-                np.reshape(baseline_tensor, tensor_spec.shape)
+                baseline_tensor = np.reshape(baseline_tensor, tensor_spec.shape)
+                baseline_sample_paths.append(baseline_tensor_path)
```
[the remark this answers is not in the document]
Comment (StrycekSimon) [reply backends/nxp/tests_models/model_output_comparator.py:176]: The shape should be consistent. It is also done this way in other comparators.

**Review by robert-kalmar — APPROVED** [review]
